src/Core/src/Platform/Android/SoftInputExtensions.cs

```diff
@@ -0,0 +1,74 @@
+﻿using Android.Content;
+using Android.Text;
+using Android.Views.InputMethods;
+using Android.Views;
+using Android.Widget;
+using AndroidX.Core.View;
+using AView = Android.Views.View;
+
+namespace Microsoft.Maui.Platform
+{
+	public static partial class SoftInputExtensions
+	{
+		internal static bool HideSoftInput(this AView inputView)
+		{
+			AView tokenView = inputView;
+
+			using var inputMethodManager = (InputMethodManager?)tokenView.Context?.GetSystemService(Context.InputMethodService);
+			var windowToken = tokenView.WindowToken;
+
+			if (windowToken is not null && inputMethodManager is not null)
+			{
+				return inputMethodManager.HideSoftInputFromWindow(windowToken, HideSoftInputFlags.None);
+			}
+
+			return false;
+		}
+
+		internal static bool ShowSoftInput(this TextView inputView)
+		{
+			using var inputMethodManager = (InputMethodManager?)inputView.Context?.GetSystemService(Context.InputMethodService);
+
+			// The zero value for the second parameter comes from 
+			// https://developer.android.com/reference/android/view/inputmethod/InputMethodManager#showSoftInput(android.view.View,%20int)
+			// Apparently there's no named value for zero in this case
+			return inputMethodManager?.ShowSoftInput(inputView, 0) is true;
+		}
+
+		internal static bool ShowSoftInput(this AView view) => view switch
```

Comment:
@PureWeen wrote the original code and could speak more to this implementation, but I think what's happening is that for a given Android `View`, if it's a TextView then it will call the given `ShowSoftInput` on it to bring up the keyboard. If it's a group, then it will search within it to find any TextView that can be focused to show the input. If there are none, it returns false.